### DISCUSSION TITLE
Trying to get a deeper understanding of how verification works

### DISCUSSION THREAD

**Discussion by quintessence** [mastodon/mastodon#21192] [Q&A]
Hello!

We've been writing up some user docs over at Hachyderm and we were trying to explain some of the nuance with verification, especially since some sites (notably GitHub) have special instructions.

A question I had from all of that: when is verification checked? Is it like a heartbeat, where it's checked periodically or is it only checked once by the person who accesses your profile?

There seems to be some disagreement on the web whether you need to leave the URL on a third party site (your .com, wherever) permanently or not. For transparency, this is mostly coming up because we have users interested in using GitHub and it seems like the primary way to do that is to use the website URL on the profile field, which uses up that field permanently unless users can add, verify, and revert to however they were using it previously.

**Comment by sukima** [comment]
I'd like to add also about how it scans the page for the verification. All the docs say as long as the `rel="me"` and `href="…"` remains you can do what you like. My question is does the scan only look for `[rel=me][href]` or `a[rel=me][href]`. Can I instead add this to the head of my HTML?

```html
<link rel="me" href="…">
```

**Reply by trwnh** [reply]
yes, `<a>` and `<link>` are both acceptable

**Reply by nightpool** [reply]
a and link elements work but other elements do not

**Comment by quintessence** [comment]
My "I am not a Mastodon maintainer but based on our experience" answer is that the `<link rel="me" href="..">` in the header _does_ seem to work.

That said: we did notice some inconsistencies for users _staying_ verified, which we're unsure the cause of. In my case I asked some of our users to tell me if my GitHub appeared verified in their various sessions and most web users said yes, some inconsistency from the different mobile apps. No error codes to follow up on, and the combination of not being sure why the verification inconsistently works or when it does completely fail is mostly what prompted my/our request 😅

**Comment by trwnh** [comment]
> There seems to be some disagreement on the web whether you need to leave the URL on a third party site (your .com, wherever) permanently or not.

Yes, it needs to be there perpetually, because this two-way relation is checked by every remote instance whenever the profile is discovered or updated. It is also checked locally whenever you edit your profile fields.

**Comment by nightpool** [comment]
> A question I had from all of that: when is verification checked? Is it like a heartbeat, where it's checked periodically or is it only checked once by the person who accesses your profile?

verification is checked by each server individually. I think it happens only once for each new link, but it may happen redundantly when the profile is updated. I haven't looked at that code recently.

That means that unless you remove the link from your profile, every new server that fetches your profile will also check your verification. This happens all the time as new servers join the network, people follow you, you reply to new people or your posts get boosted, etc

**Comment by quintessence** [comment]
Quick question: I had some of our users (Hachyderm) in an informal poll try to see if my GitHub was showing as "verified" or not, and there seemed to be inconsistency there even though my Mastodon URL is in my GitHub web URL field. I've heard of similar with different sites using the verification method where the verification might appear to show as "valid" on a user's individual session but not for the remote checks, even if the URL is still there to be verified by the remote server. Do different apps / etc. check for verification differently?
